opslevel/resource_opslevel_trigger_definition.go

```diff
@@ -106,6 +114,48 @@ func (r *TriggerDefinitionResource) Schema(ctx context.Context, req resource.Sch
 				Required:    true,
 				Validators:  []validator.String{IdStringValidator()},
 			},
+			"approval_config": schema.MapNestedAttribute{
+				Description: "The configuration defining conditions of approval if it is required.",
+				Optional:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"approval_required": schema.BoolAttribute{
+							Description: "Flag indicating approval is required.",
+							Optional:    true,
+							Computed:    true,
+							Default:     booldefault.StaticBool(false),
+						},
+						"teams": schema.ListNestedAttribute{
+							Description: "Teams that can approve.",
+							Optional:    true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"alias": schema.StringAttribute{
+										Required: true,
+									},
+									"id": schema.StringAttribute{
+										Required: true,
+									},
+								},
+							},
+						},
```

Comment:
The way we've handled "IdentifierInput" is that we have a special function in opslevel-go that can make a new `IdentifierInput` based on if the string is detected as an ID - then it fills out the ID field - and if not it fills out the alias field.  This means the customer can just use a list of strings like this
```tf
  teams = ["098sdflkjwerlkjwertkljsdf90sdf908", "my-team-alias", "my-team-alias2"]
```

So i think this can just be a `ListAttribute` of strings.  This provides a much better UX to the customer.

Same goes for the `users` field.


Where this gets tricky on our side is that we need to shave into the state exactly what they gave us instead of what our API returned.  We usually do this as special case logic in the  "NewModel" function.